ratis-common/src/main/java/org/apache/ratis/util/Daemon.java

```diff
@@ -37,4 +57,32 @@ public Daemon(Runnable runnable, String name) {
     super(runnable);
     this.setName(name);
   }
+
+  public Daemon(Runnable runnable, String name, ErrorRecorded server) {
```

Comment:
I ignored `Daemon` threads created in `JvmPauseMonitor` and `TimeoutScheduler` because my intuition is even if they crash, the RaftServer can still live.